### ISSUE TITLE
The "padding" parameter of tokenizer doesn't seem to work

### ISSUE THREAD

**Issue by a815063199** [huggingface/transformers#26131]
### System Info

- `transformers` version: 4.33.0
- Platform: Linux-4.15.0-206-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- Huggingface_hub version: 0.16.4
- Safetensors version: 0.3.3
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@ArthurZucker
The padding parameter of the tokenizer seems to have no effect, and the tensor shape returned by the tokenizer using padding is not being padded. Use pad_to_max_length parameter can be correctly padding

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```python
from transformers import BertModel, BertTokenizer, BertTokenizerFast

model_name = 'moka-ai/m3e-base'
tokenizer = BertTokenizerFast.from_pretrained(model_name)

sentences = "我今天中午吃的拉面，你吃的啥"

inputs = tokenizer.encode_plus(
    sentences,                           # 输入文本
    add_special_tokens=True,        # 添加特殊标记
    max_length=512,                 # 最大长度
    #pad_to_max_length=True,          # 使用padding填充
    padding=True,          # 使用padding填充
    truncation=True,
    return_attention_mask=True,      # 返回attention mask
    return_tensors='pt'              # 返回pt类型的张量
)

for k, v in inputs.items():
    print(v.shape) # except shape is (1, 512), but get (1, 16) when using "padding=True"
```

### Expected behavior

the padding parameter of tokenizer should work same as the pad_to_max_length

**Comment by a815063199** [comment]
ok～ i misunderstood the meaning of `padding=true`. thanks`
